includes/admin/importers/class-wc-product-csv-importer-controller.php

```diff
@@ -8,83 +8,180 @@
 
 	/**
-	 * The current file id.
+	 * The path to the current file.
 	 *
-	 * @var int
+	 * @var string
 	 */
-	public $id;
+	protected $file = '';
 
 	/**
-	 * The current file url.
+	 * The current import step.
 	 *
 	 * @var string
 	 */
-	public $file_url;
+	protected $step = '';
 
 	/**
-	 * The current import page.
+	 * Progress steps.
 	 *
-	 * @var string
+	 * @var array
+	 */
+	protected $steps = array();
+
+	/**
+	 * Errors.
+	 *
+	 * @var array
 	 */
-	public $import_page;
+	protected $errors = array();
 
 	/**
-	 * The current delimiter.
+	 * The current delimiter for the file being read.
 	 *
 	 * @var string
 	 */
-	public $delimiter;
+	protected $delimiter = ',';
 
 	/**
 	 * Constructor.
 	 */
 	public function __construct() {
-		$this->import_page = 'woocommerce_product_csv';
-		$this->delimiter   = empty( $_REQUEST['delimiter'] ) ? ',' : (string) wc_clean( $_REQUEST['delimiter'] );
+		$this->steps = array(
+			'upload' => array(
+				'name'    => __( 'Upload CSV file', 'woocommerce' ),
+				'view'    => array( $this, 'upload_form' ),
+				'handler' => array( $this, 'upload_form_handler' ),
+			),
+			'mapping' => array(
+				'name'    => __( 'Column mapping', 'woocommerce' ),
+				'view'    => array( $this, 'mapping_form' ),
+				'handler' => '',
+			),
+			'import' => array(
+				'name'    => __( 'Import', 'woocommerce' ),
+				'view'    => array( $this, 'import' ),
+				'handler' => '',
+			),
+			'done' => array(
+				'name'    => __( 'Done!', 'woocommerce' ),
+				'view'    => array( $this, 'done' ),
+				'handler' => '',
+			),
+		);
+		$this->step = isset( $_REQUEST['step'] ) ? sanitize_key( $_REQUEST['step'] ) : current( array_keys( $this->steps ) );
+		$this->file = isset( $_REQUEST['file'] ) ? wc_clean( $_REQUEST['file'] ) : '';
 	}
 
 	/**
-	 * Registered callback function for the WordPress Importer.
-	 *
-	 * Manages the three separate stages of the CSV import process.
+	 * Get the URL for the next step's screen.
+	 * @param string step   slug (default: current step)
+	 * @return string       URL for next step if a next step exists.
+	 *                      Admin URL if it's the last step.
+	 *                      Empty string on failure.
 	 */
-	public function dispatch() {
-		include_once( WC_ABSPATH . 'includes/import/class-wc-product-csv-importer.php' );
+	public function get_next_step_link( $step = '' ) {
+		if ( ! $step ) {
+			$step = $this->step;
+		}
 
-		$this->header();
+		$keys = array_keys( $this->steps );
 
-		$step = empty( $_GET['step'] ) ? 0 : (int) $_GET['step'];
+		if ( end( $keys ) === $step ) {
+			return admin_url();
+		}
 
-		switch ( $step ) {
+		$step_index = array_search( $step, $keys );
 
-			case 0:
-				$this->greet();
-				break;
+		if ( false === $step_index ) {
+			return '';
+		}
 
-			case 1 :
-				check_admin_referer( 'import-upload' );
+		$params = array(
+			'step'      => $keys[ $step_index + 1 ],
+			'file'      => $this->file,
+			'file_url'  => $this->file_url,
```

Comment:
This line threw the following error in my URL bar because you removed `$file_url` on line 32.

`
http://local.wordpress.dev/wp-admin/%3Cbr%20/%3E%3Cb%3ENotice%3C/b%3E:%20%20Undefined%20property:%20WC_Product_CSV_Importer_Controller::$file_url%20in%20%3Cb%3E/srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/admin/importers/class-wc-product-csv-importer-controller.php%3C/b%3E%20on%20line%20%3Cb%3E112%3C/b%3E%3Cbr%20/%3E/wp-admin/edit.php?post_type=product&page=product_importer&step=import&file=%2Fsrv%2Fwww%2Fwordpress-default%2Fpublic_html%2Fwp-content%2Fuploads%2F2017%2F05%2Fcsvimporter-testing.csv-1.txt&delimiter=%2C&_wpnonce=3ad5513515
`